POEM_081.md

```diff
@@ -0,0 +1,421 @@
+
+Furthermore, it has the added benefit of allowing for nested drivers. The OpenMDAO system within a subproblem can be subject to constraints, have design variables, and objectives at its system level. This makes implementing nested drivers more intuitive for users.
+
+The code below is how the new subproblem component will look. Its inputs are similar to that of `Problem`, and since it is an `om.ExplicitComponent`, it can be added as a subsystem to a model in a top-level problem.
+
+## Prototype Implementation
+
+```python
+import openmdao.api as om
+from openmdao.core.constants import _UNDEFINED
+from openmdao.utils.om_warnings import issue_warning
+
+
+class SubproblemComp(om.ExplicitComponent):
+    """
+    System level container for systems and drivers.
+    
+    Parameters
+    ----------
+    model : <System> or None
+        The top-level <System>. If not specified, an empty <Group> will be created.
+    driver : <Driver> or None
+        The driver for the problem. If not specified, a simple "Run Once" driver will be used.
+    comm : MPI.Comm or <FakeComm> or None
+        The global communicator.
+    name : str
+        Problem name. Can be used to specify a Problem instance when multiple Problems
+        exist.
+    reports : str, bool, None, _UNDEFINED
+        If _UNDEFINED, the OPENMDAO_REPORTS variable is used. Defaults to _UNDEFINED.
+        If given, reports overrides OPENMDAO_REPORTS. If boolean, enable/disable all reports.
+        Since none is acceptable in the environment variable, a value of reports=None
+        is equivalent to reports=False. Otherwise, reports may be a sequence of
+        strings giving the names of the reports to run.
+    prob_kwargs : dict
+        All remaining args typically used in a problem. Must be stored in a dict since
+        **kwargs is used for om.ExplicitComponent instead.
+    inputs : list of str or tuple
+        List of desired inputs to subproblem. If an element is a str, then it should be
+        the var name in its promoted name. If it is a tuple, then the first element 
+        should be the promoted name, and the second element should be the var name
+        you wish to refer to it by within the subproblem [e.g. (prom_name, var_name)].
+    outputs : list of str or tuple
+        List of desired outputs from subproblem. If an element is a str, then it should be
+        the var name in its promoted name. If it is a tuple, then the first element 
+        should be the promoted name, and the second element should be the var name
+        you wish to refer to it by within the subproblem [e.g. (prom_name, var_name)].
+    **kwargs : named args
+        All remaining named args that become options for `SubproblemComp`.
+    """
+    def __init__(self, model=None, driver=None, comm=None, name=None,
```

Comment:
I'm curious why this component takes a model and some Problem args, then constructs an internal problem (twice) instead of just passing a Problem instance and the list of inputs and outputs into it.  I could be missing something, but it seems like that would be a much simpler API.

Reply:
You're right. My initial thought was to take away the steps of instantiating a problem from the user, but looking at what I have, it would definitely be easier to instantiate a problem outside of subproblem and pass that in. Also, I guess I didn't catch in the moment that the two instances of problem wasn't the most effective way of doing this, so thanks for pointing that out.
I want model to be passed as an argument along with problem too, however. This way, that extra step of adding the model as a subsystem to the problem model is handled and there is no confusion with that.
Thanks for looking through this, I will update the POEM soon.